src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -65,12 +65,46 @@ private function completeDefinition($id, Definition $definition)
 
         $this->container->addClassResource($reflectionClass);
 
-        if (!$constructor = $reflectionClass->getConstructor()) {
+        if ($constructor = $reflectionClass->getConstructor()) {
+            $this->autowireMethod($id, $definition, $constructor, true);
+        }
+
+        $methodsCalled = array();
+        foreach ($definition->getMethodCalls() as $methodCall) {
+            $methodsCalled[$methodCall[0]] = true;
+        }
+
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
+            $name = $reflectionMethod->getName();
+            if (isset($methodsCalled[$name]) || $reflectionMethod->isStatic() || 1 !== $reflectionMethod->getNumberOfParameters() || 0 !== strpos($name, 'set')) {
+                continue;
+            }
+
+            $this->autowireMethod($id, $definition, $reflectionMethod, false);
+        }
+    }
+
+    /**
+     * Autowires the constructor or a setter.
+     *
+     * @param string            $id
+     * @param Definition        $definition
+     * @param \ReflectionMethod $reflectionMethod
+     * @param bool              $constructor
+     *
+     * @throws RuntimeException
+     */
+    private function autowireMethod($id, Definition $definition, \ReflectionMethod $reflectionMethod, $constructor)
+    {
+        if ($constructor) {
+            $arguments = $definition->getArguments();
+        } elseif (0 === $reflectionMethod->getNumberOfParameters()) {
```

Comment:
This test is useless (won't change anything for the constructor and already checked earlier for the setters).